### PR TITLE
Preferences > Key: restore custom notation

### PR DESCRIPTION
**Pull request by ronso0**
reported in the forums
https://mixxx.discourse.group/t/custom-key-notation-not-saved-after-restart-in-2-3-0/

custom key notation was not restored correctly.
code is simpler now, too.

## Proposed Changelog entry

    - Fix custom key notation not restored correctly after restart #4136

### CODE REVIEW

**Review by Swiftb0y — COMMENTED** [review]
I don't fully understand the code changes to be honest, but I was able to confirm that the issue exists in main and 2.3 and is fixed by this PR.
Do yo want to include a changelog entry? I don't know our current policy on what to include in the changelog when.

File: src/preferences/dialog/dlgprefkey.cpp [hunk 1]
```diff
@@ -265,25 +265,20 @@ void DlgPrefKey::slotUpdate() {
     }
 }
 
-void DlgPrefKey::setNotationCustom(bool active) {
-    if (!active) {
-        return;
-    }
-
+void DlgPrefKey::setNotation(KeyUtils::KeyNotation notation) {
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefkey.cpp:268]: The whole class and especially these methods are not so great. But at least the handling is consistent now.

File: src/preferences/dialog/dlgprefkey.cpp [hunk 2]
```diff
@@ -265,25 +265,20 @@ void DlgPrefKey::slotUpdate() {
...
     for (auto it = m_keyLineEdits.constBegin();
             it != m_keyLineEdits.constEnd(); ++it) {
-        it.value()->setEnabled(true);
+        it.value()->setText(KeyUtils::keyToString(it.key(), notation));
+        // QLineEdits are only enabled for Custon notation.
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefkey.cpp:272]: typo

**Review by uklotzde — COMMENTED** [review]
Only a minor typo.

File: src/preferences/dialog/dlgprefkey.cpp [hunk 3]
```diff
@@ -128,8 +128,8 @@ void DlgPrefKey::loadSettings() {
         }
     }
 
-    setNotation(notation_type);
     KeyUtils::setNotation(notation);
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefkey.cpp:131]: What was the reason for swapping the invocations? Then maybe add a comment that explains why the order is relevant.

Comment (ronso0) [reply src/preferences/dialog/dlgprefkey.cpp:132]: `KeyUtils::setNotation(notation)` stores the (custom) notation map,
and in `setNotation()` the QLineEdits are populated from KeyUtils.
the wrong order was one reason the initial population was done from a default map.
Comment added, squashed, rebased

File: src/preferences/dialog/dlgprefkey.cpp (hunk 2 above)
Comment (ronso0) [reply src/preferences/dialog/dlgprefkey.cpp]: thx, fixed

**Comment by ronso0** [comment]
ready.

File: src/preferences/dialog/dlgprefkey.cpp (hunk 3 above)
Comment (uklotzde) [reply src/preferences/dialog/dlgprefkey.cpp:132]: Thank you. This is another example why code with hidden side effects must strictly be avoided.

File: src/preferences/dialog/dlgprefkey.cpp [hunk 4]
```diff
@@ -128,8 +128,11 @@ void DlgPrefKey::loadSettings() {
         }
     }
 
-    setNotation(notation_type);
+    // Store notation map for later rećall...
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefkey.cpp:131]: Why all the accents? Maybe one of your keys got stuck ;)

File: src/preferences/dialog/dlgprefkey.cpp [hunk 5]
```diff
@@ -128,8 +128,11 @@ void DlgPrefKey::loadSettings() {
...
     KeyUtils::setNotation(notation);
+    // ... BEFORE invóking setNotation() which populates the QLineEdits from
+    // the map retreived from KeyUtils.
```
Comment (uklotzde) [inline src/preferences/dialog/dlgprefkey.cpp:134]: and a typo

File: src/preferences/dialog/dlgprefkey.cpp (hunk 4 above)
Comment (ronso0) [reply src/preferences/dialog/dlgprefkey.cpp:131]: naa, my display now has cracks all over the left side. so usually it's 'why that accent? oh, it's just a crack..'
here it was the other way around : )

File: src/preferences/dialog/dlgprefkey.cpp (hunk 5 above)
Comment (ronso0) [reply src/preferences/dialog/dlgprefkey.cpp:134]: fixed

**Comment by ronso0** [comment]
*pa-ping*

**Comment by uklotzde** [comment]
Thank you! LGTM

**Pull request merged**